### PR TITLE
Basic DB setup

### PR DESCRIPTION
**Pull request by ttung**
### Description
It is an empty schema with the most rudimentary schema migration support.

Depends on #7 

#### Issue
[ch65314](https://app.clubhouse.io/genepi/stories/space/65314)

### Test plan

1. `make init-local-db` and `make drop-local-db` both work.
2. Made some dummy changes to the model code and ran `ENV=local alembic revision --autogenerate -m "SOME DESCRIPTIVE MESSAGE" --rev-id $(date +%Y%M%d_%H%M%S)` to generate a migration.  Hand checked it to see that it made sense.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,62 @@
...
+	ENV=local alembic stamp head
+
+stop-local-db:
+	@if [ "$(LOCAL_DB_CONTAINER_RUNNING_ID)" != "" ]; then \
+		docker stop $(LOCAL_DB_CONTAINER_NAME); \
+	fi
+
+drop-local-db:
+	@$(MAKE) stop-local-db
+	@( read -p "Delete local database container? [y/N]: " sure && case "$$sure" in [yY]) true;; *) false;; esac )
```
Comment (phoenixAja) [inline Makefile:51]: it's throwing me off that this @ sign is red, do you know why it's doing that?

Comment (phoenixAja) [reply Makefile:51]: i looked at the same line in the covidhub Makefile and it looks the same so i think this is correct, i was jw

File: src/py/covidr/cli/db.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import click
+from covidr.database.connection import get_db_uri, init_db
+from covidr.database.schema import create_tables_and_schema
+from covidr.sandbox.runtimeenvironment import RuntimeEnvironment
+
+from .toplevel import cli
+
+
+@cli.group()
```
Comment (phoenixAja) [inline src/py/covidr/cli/db.py:9]: interesting, going with click instead of argparse?

File: src/py/covidr/database/modelmixins.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+
+# for things that have names
+class NameMixin:
+    """Mixin that adds a required unique name string field"""
+
+    name = Column(String, unique=True, nullable=False)
+
+    def __repr__(self):
+        return self.name
```
Comment (phoenixAja) [inline src/py/covidr/database/modelmixins.py:20]: i'm confused when we would need this over the `BaseMixin`? Like the primary_key will always be unique so why would we need this?

Comment (phoenixAja) [reply src/py/covidr/database/modelmixins.py:20]: oh nevermind i get it the comment was throwing me off

File: src/py/covidr/database/models.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# the main models in the database
+from sqlalchemy import MetaData
+from sqlalchemy.ext.declarative import declarative_base
+
+from . import modelmixins as mx
+
+meta = MetaData(
+    schema="covidr",
```
Comment (phoenixAja) [inline src/py/covidr/database/models.py:8]: do we need to create this schema with a migration?

File: src/py/covidr/database/schema.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+
+def create_tables_and_schema(interface: SqlAlchemyInterface):
+    """Creates the schema and all tables"""
+    with session_scope(interface) as session:
+        sqlalchemy_event.listen(
+            meta,
+            "before_create",
+            DDL(f"CREATE SCHEMA IF NOT EXISTS {meta.schema}"),
+        )
```
Comment (phoenixAja) [inline src/py/covidr/database/schema.py:15]: oh ok interesting, why not just create the schema in the initial migration?

**Review by phoenixAja — APPROVED** [review]
looks good, just had a couple questions

File: Makefile (hunk 1 above)
Comment (ttung) [reply Makefile:51]: the `@` silences the echoing of the command when it's run.  it's mostly for aesthetic reasons.  Not sure why it's highlighted red.

File: src/py/covidr/cli/db.py (hunk 2 above)
Comment (ttung) [reply src/py/covidr/cli/db.py:9]: I go back and forth. :)

File: src/py/covidr/database/schema.py (hunk 5 above)
Comment (ttung) [reply src/py/covidr/database/schema.py:15]: The reason is that I want `make init-local-db` actually produce a functional db.

**Pull request merged**